Sources/_NIOFileSystem/FileSystem.swift

```diff
@@ -1311,7 +1329,7 @@ extension FileSystem {
                             location: .here()
                         )
                     }
-
+                    #endif
```

Comment:
Let's factor this `#if/#else` into a helper function to avoid it making this code too complex.